WordPress/gutenberg
# Block Support: Update border support UI

Part of: https://github.com/WordPress/gutenberg/issues/31337
Related: https://github.com/WordPress/gutenberg/pull/32080
~Depends on: #31483 & https://github.com/WordPress/gutenberg/pull/31641~

## Description

This PR refines the border block support UI in line with https://github.com/WordPress/gutenberg/issues/31337. 

These refinements are separate to migrating the border block support to use the new progressive display panel proposed in https://github.com/WordPress/gutenberg/pull/32392. Converting to this new panel will be best handled separately.

The condensed color control proposed in the G2 components still requires some additional dependencies being imported into Gutenberg so will also be addressed in a future PR as soon as the control is available.

## How has this been tested?
- Manually in the editor. 
- Updated phpunit test for stylesheet generation

#### Test instructions
1. Checkout this PR branch and build
2. Enable border support UI via theme.json - [example theme.json gist](https://gist.github.com/aaronrobertshaw/8ed108e4a024b81297a55fc498ef3a57)
3. Add group block to a post, select it
3. Confirm border support controls display and function as expected

## Screenshots 
![UpdatedBorderSupportUI](https://user-images.githubusercontent.com/60436221/117418310-9cf4af00-af5e-11eb-9cd5-3f7e5655b2f9.gif)


## Types of changes
New feature. 

## Checklist:
